### ISSUE TITLE
`readability-non-const-parameter`:  false positive for placement-new storage

### ISSUE THREAD

**Issue by MatzeB** [llvm/llvm-project#206613]
For a pointer parameter used only as storage for a placement-new expression,
  `readability-non-const-parameter` suggests making it pointer-to-const. The
  pointee *is* modified (placement-new constructs an object into it), so the
  parameter must stay non-const. Applying the fixit produces code that no longer
  compiles — and `clang-tidy --fix` applies it automatically.

```cpp
#include <new>

  struct S { int x; };

  void construct(char *buffer) {
    new (buffer) S();
  }
```

Produces:
```
$ clang-tidy --checks='-*,readability-non-const-parameter' repro.cpp -- -std=c++17
warning: pointer parameter 'buffer' can be pointer to const
  [readability-non-const-parameter]
      6 | void construct(char *buffer) {
        |                      ^
        |                const
```

and applying auto-fixup breaks the source. Somehow `void*` parameters are not affected.

Tested with clang-tidy-22 did not try newer versions or trunk

**Comment by vbvictor** [comment]
https://godbolt.org/z/5TP68caG3
